go/arrow/extensions/bool8.go

```diff
@@ -0,0 +1,196 @@
+package extensions
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/apache/arrow/go/v17/arrow"
+	"github.com/apache/arrow/go/v17/arrow/array"
+	"github.com/apache/arrow/go/v17/arrow/memory"
+	"github.com/apache/arrow/go/v17/internal/json"
+)
+
+const ExtensionNameBool8 = "bool8"
+
+// Bool8Type represents a logical boolean that is stored using 8 bits.
+type Bool8Type struct {
+	arrow.ExtensionBase
+}
+
+func NewBool8Type() *Bool8Type {
+	return &Bool8Type{ExtensionBase: arrow.ExtensionBase{Storage: arrow.PrimitiveTypes.Int8}}
+}
+
+func (b *Bool8Type) ArrayType() reflect.Type { return reflect.TypeOf(Bool8Array{}) }
+
+func (b *Bool8Type) Deserialize(storageType arrow.DataType, data string) (arrow.ExtensionType, error) {
+	if data != ExtensionNameBool8 {
+		return nil, fmt.Errorf("type identifier did not match: '%s'", data)
+	}
+	if !arrow.TypeEqual(storageType, arrow.PrimitiveTypes.Int8) {
+		return nil, fmt.Errorf("invalid storage type for Bool8Type: %s", storageType.Name())
+	}
+	return NewBool8Type(), nil
+}
+
+func (b *Bool8Type) ExtensionEquals(other arrow.ExtensionType) bool {
+	return b.ExtensionName() == other.ExtensionName()
+}
+
+func (b *Bool8Type) ExtensionName() string { return ExtensionNameBool8 }
+
+func (b *Bool8Type) Serialize() string { return ExtensionNameBool8 }
+
+func (b *Bool8Type) String() string { return fmt.Sprintf("Bool8<storage=%s>", b.Storage) }
```

Comment:
convention is all lower case

I also think that convention is `extension_type<storage=%s>` for the String method

Reply:
It looks like that's exactly what the `ExtensionBase` will fall back to if this is not implemented:
```go
func (e *ExtensionBase) String() string { return fmt.Sprintf("extension_type<storage=%s>", e.Storage) }
```

Is there a convention for overriding this when the actual extension type is known, or should it just fall back to the default?

Reply:
I think the best would be to check the C++ lib implementations and see what they do. Ideally keeping the string representations close would be great

Reply:
The convention in C++ (current precedent is just FixedShapeTensor for now) appears to be `extension<EXTENSION_NAME[TYPE_PARAM_NAME=TYPE_PARAM_VALUE, ...]>`.

For FixedShapeTensor that ends up looking like `extension<arrow.fixed_shape_tensor[value_type=int32, shape=[2,2]]>` for example.

So then I guess in this case it would be `extension<arrow.bool8>`?